### PR TITLE
SDI-603: ✨ Add gender and ethnicity searches

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt [hunk 1]
```diff
@@ -92,26 +92,42 @@ class PhysicalDetailService(
...
   private fun buildDetailQuery(detailRequest: PhysicalDetailRequest): BoolQueryBuilder {
-    val detailQuery = boolQuery()
-
-    detailQuery.should(QueryBuilders.matchAllQuery())
+    val detailQuery = boolQuery().also {
+      it.should(QueryBuilders.matchAllQuery())
+    }
 
     with(detailRequest) {
-      minHeight?.let { detailQuery.must(boolQuery().should(rangeQuery("heightCentimetres").gte(it))) }
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PhysicalDetailService.kt:100]: just moved queries around

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt [hunk 2]
```diff
@@ -20,15 +24,8 @@ class PhysicalDetailResourceTest : AbstractSearchDataIntegrationTest() {
   @Test
   fun `access forbidden when no role`() {
     webTestClient.post().uri("/physical-detail")
-      .body(
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt:23]: turns out that `bodyValue()` is a much easier way to insert the request

File: src/test/resources/mappings/getBooking_A7090AD.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
         "gender": "Not Known / Not Recorded",
         "raceCode": "W1",
         "ethnicity": "White: Eng./Welsh/Scot./N.Irish/British",
-        "sexCode": "UK"
+        "sexCode": "NK"
```
Comment (petergphillips) [inline src/test/resources/mappings/getBooking_A7090AD.json:30]: not used, but annoyed me that the code was incorrect

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/QueueIntegrationTest.kt [hunk 4]
```diff
@@ -136,7 +138,8 @@ abstract class QueueIntegrationTest : IntegrationTest() {
 
   fun loadPrisoners(prisoner: List<PrisonerBuilder>) {
     setupIndexes()
-    val prisonerNumbers = prisoner.map { it.prisonerNumber }.toList()
+    val prisonerNumbers = prisoner.map { it.prisonerNumber }
+    assertThat(prisonerNumbers.groupingBy { it }.eachCount().filter { it.value != 1 }).hasSize(0)
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/QueueIntegrationTest.kt:142]: something to stop accidentally adding multiple prisoners with same prisoner number

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**